presto-main/src/main/java/com/facebook/presto/operator/scalar/MapSubsetFunction.java

```diff
@@ -36,8 +36,8 @@ public static Block mapSubset(
             @TypeParameter("MAP(K,V)") Type mapType,
             @SqlType("MAP(K,V)") Block mapBlock, @SqlType("ARRAY(K)") Block keySubset)
     {
-        if (mapBlock.getPositionCount() == 0 || keySubset.getPositionCount() == 0) {
```

Comment:
Looks like that we are going to iterate through the whole map trying to find match when `keySubset.getPositionCount() == 0` after the change. Can we add code to skip it?

To skip this part of code:
```
        BlockBuilder blockBuilder = mapBlockBuilder.beginBlockEntry();
        for (int i = 0; i < mapBlock.getPositionCount(); i += 2) {
            if (typedSet.contains(mapBlock, i)) {
                keyType.appendTo(mapBlock, i, blockBuilder);
                valueType.appendTo(mapBlock, i + 1, blockBuilder);
                toFind--;
                if (toFind == 0) {
                    break;
                }
            }
        }
        mapBlockBuilder.closeEntry();
```

Reply:
Yeah. If the intention is to return original map incase map is empty, then a separate check for it seems fine. However the earlier check for checking key set seems fine to me. 

Also should we add a test for empty keys as well?

Reply:
For key empty, we can return original map. We need to return empty map which is a pain to build by hand

Reply:
And test exists for empty subset array

Reply:
Nah - that;s unnecessary optimization.

Reply:
OK fixed this now I see toFind == 0 won't hit in the empty subset case.